tests/TestKeePass2XmlReader.cpp

```diff
@@ -459,19 +533,19 @@ void TestKeePass2XmlReader::testInvalidXmlChars()
 
     QBuffer buffer;
     buffer.open(QIODevice::ReadWrite);
-    KeePass2XmlWriter writer;
-    writer.writeDatabase(&buffer, dbWrite.data());
-    QVERIFY(!writer.hasError());
+    bool hasError;
+    QString errorString;
+    writeDatabase(&buffer, dbWrite.data(), hasError, errorString);
+    QVERIFY(!hasError);
     buffer.seek(0);
 
-    KeePass2XmlReader reader;
-    reader.setStrictMode(true);
-    QScopedPointer<Database> dbRead(reader.readDatabase(&buffer));
-    if (reader.hasError()) {
-        qWarning("Database read error: %s", qPrintable(reader.errorString()));
+    Database* dbRead;
```

Comment:
You can use the dereference operator on a QScopedPointer to get a reference to the guarded raw pointer. No need to revert back here.